__tests__/Car.test.js

```diff
@@ -1,49 +1,82 @@
-import RacingCarGame from "../src/class/RacingCarGame";
+import Validator from "../src/class/Validator";
+import Cars from "../src/class/Cars";
+import Car from "../src/class/Car";
+
+describe("Validator Class 테스트", () => {
+  describe("validateCarNames 테스트", () => {
+    test("자동차 이름 중 빈값인 자동차가 포함되면 에러가 발생한다.", () => {
+      expect(() => Validator.validateCarNames(["", "테스트"])).toThrowError(
+        "자동차 이름은 빈값일 수 없습니다.",
+      );
+    });
 
-describe("RacingCarGame Class 테스트", () => {
-  test("validateCarNames에서 에러가 발생하면 에러가 onError에 전달되어야 하며 게임이 종료된다.", async () => {
-    const expectedError = new Error("에러 테스트");
+    test("자동차 이름 중 6자 이상인 자동차가 있으면 에러가 발생한다.", async () => {
+      expect(() =>
+        Validator.validateCarNames(["테스트", "123456"]),
+      ).toThrowError("자동차 이름은 5자를 넘길 수 없습니다.");
+    });
 
-    const validateCarNamesTester = () => {
-      throw expectedError;
-    };
+    test("자동차 이름 중 같은 이름의 자동차가 있으면 에러가 발생한다.", async () => {
+      expect(() =>
+        Validator.validateCarNames(["테스트", "테스트"]),
+      ).toThrowError("자동차 이름은 중복될 수 없습니다.");
+    });
+  });
 
-    const handleErrorTester = jest.fn();
+  describe("validateRoundNumber 테스트", () => {
+    test("0~9 이외의 문자가 포함된 이동횟수 입력시 에러가 발생한다.", () => {
+      const testRoundNumber = "-0.1";
 
-    const racingCarGame = new RacingCarGame({
-      validateCarNames: validateCarNamesTester,
-      onError: handleErrorTester,
+      expect(() => Validator.validateRoundNumber(testRoundNumber)).toThrowError(
+        "양의 정수 형식의 값을 입력해 주세요.",
+      );
     });
 
-    const endGameSpy = jest.spyOn(racingCarGame, "endGame");
+    test("0이하의 숫자가 입력된 경우 에러가 발생한다", () => {
+      const testRoundNumber = "0";
 
-    const onErrorSpy = jest.spyOn(racingCarGame, "onError");
+      expect(() => Validator.validateRoundNumber(testRoundNumber)).toThrowError(
+        "1이상 값을 입력해주세요.",
+      );
+    });
+  });
+});
 
-    await racingCarGame.startGame();
+describe("Car Class 테스트", () => {
+  test("advanceCondition이 true를 반환하면 distance가 1 증가한다.", async () => {
+    const mockAdvanceCondition = () => true;
 
-    expect(validateCarNamesTester).toThrow(expectedError);
+    const car = new Car("test Car", mockAdvanceCondition);
```

Comment:
```suggestion
    const car = new Car({}, mockAdvanceCondition);
```
이런식으로 엉뚱한 값이 들어간다면 어떻게 해야 좋을까요?